packages/react-native/scripts/cocoapods/utils.rb

```diff
@@ -59,15 +59,23 @@ def self.turn_off_resource_bundle_react_core(installer)
     end
 
     def self.exclude_i386_architecture_while_using_hermes(installer)
+        isUsingHermes = self.has_pod(installer, 'hermes-engine')
+        key = "EXCLUDED_ARCHS[sdk=iphonesimulator*]"
+
         projects = self.extract_projects(installer)
 
         projects.each do |project|
             project.build_configurations.each do |config|
-                currentValue = config.build_settings["EXCLUDED_ARCHS[sdk=iphonesimulator*]"] || ""
+                current_setting = config.build_settings[key] || ""
+
+                excludesI386 = current_setting.include?("i386")
 
-                unless currentValue.include?("i386")
+                if isUsingHermes and !excludesI386
```

Comment:
```suggestion
excludesI386 = !current_setting.include?("i386")
if isUsingHermes and excludesI386
```

First, you were calling a variable `excludeXXX` when using the `include?` method, which is a bit confusing, as it negates the meaning of the method without actually negating that.

Secondly the `!exclude` bit was a double negation... which makes it harder to reason about what's happening.

Reply:
renamed to `build_will_skip_I386`